File: C/2231.c
#include <stdio.h>

int main(){

    int n, m, teste = 1;

    scanf("%i %i", &n, &m);

    while(1){
        if(n == 0 && m == 0){
            break;
        }

        int temperaturas[n], medias[n-m+1], maior = -200, menor = 200;

        for(int i = 0; i < n; i++){
            scanf("%i", &temperaturas[i]);
        }

        for(int i = 0, b = n-m+1; i < b; i++){
            int media = 0;

            for(int j = i, c = i+m; j < c; j++){
                media += temperaturas[j];
            }

            media = media / m;
            medias[i] = media;
            
            if(media < menor){
                menor = media;
            }

            if(media > maior){
                maior = media;
            }
        }

        printf("Teste %i\n", teste);
        printf("%i %i\n", menor, maior);
        printf("\n");

        scanf("%i %i", &n, &m);
        teste++;
    }

    return 0;
}

File: Python/2231.py
n, m = map(int, input().split())

teste = 1

while n != 0 and m != 0:
    
    temperaturas = []
    medias = []
    
    for i in range(n):
        temperaturas.append(int(input()))
        
    for i in range(n - m + 1):
        media = int(sum(temperaturas[i: i+m]) / m)
        medias.append(media)
        
    maior = max(medias)
    menor = min(medias)
            
    print(f"Teste {teste}")
    print(f"{menor} {maior}")
    print()
    
    print(medias)
    
    n, m = map(int, input().split())
    teste += 1